src/System.Management.Automation/engine/regex.cs

```diff
@@ -231,13 +235,17 @@ internal static string Escape(string pattern, char[] charsNotToEscape)
 
             string s = null;
 
-            if (tempIndex > 0)
+            if (tempIndex == 0)
```

Comment:
This is basically `Pattern == string.Emtpy`. It should be checked before allocating the buffer, so we don't create a 1-char buffer for it.

Reply:
Fixed.